gclient/tests/client.rs

```diff
@@ -0,0 +1,61 @@
+// This file is part of Gear.
+
+// Copyright (C) 2022-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use anyhow::Result;
+use demo_proxy::{InputArgs, WASM_BINARY};
+use gclient::{
+    client::{Backend, Client, GClient, GTest, Message},
+    GearApi,
+};
+
+async fn test_ping<T: Backend>(client: Client<T>) -> Result<()> {
+    let prog = client
+        .deploy(
+            WASM_BINARY,
+            InputArgs {
+                destination: [0; 32],
+            },
+        )
+        .await?
+        .result;
+
+    let ping = b"ping";
+    let result = prog.send(Message::bytes(ping)).await?;
+
+    assert!(
+        result
+            .logs
```

Comment:
I think it's better to expose  events as a Stream (in Backand trait). We did it in sails for gclient & gtest
https://github.com/gear-tech/sails/blob/master/rs/src/gsdk/calls.rs#L125
https://github.com/gear-tech/sails/blob/master/rs/src/gtest/calls.rs#L171

For gtest we collect all events from RunResult and send them over the channel
https://github.com/gear-tech/sails/blob/master/rs/src/gtest/calls.rs#L89

Reply:
> Generally speaking result.log way to far from being a convenient abstraction. It would be better if there is a straight option to get response as a payload, error or whatever + there is an option to sbscribe to events.

logs are more frequently used than response in production

1. for a DEX example, user send request to swap 5 USD for 1 DOT, logging the result `Alice swapped 1 DOT with 5 USD` is cheaper than sending the user a message with the `result`
2. however, the `send <-> reply` model is for program integrations, since the `reply` need to pay for staying on chain, it will be logged as well for saving money


> I think it's better to expose events as a Stream (in Backand trait). 

if I'm not mistaken,  this is for unifying the `subscribe` method of gtest & gclient, imo we don't need to implement the `subscribe / listen` method for `gtest` since it is not for tests )))

developers can just verify if their logic works as expected and checking the logs of each calls, otherwise they need to use the gclient `Client` for setting up their integration tests, testnet, etc.